packages/flutter/lib/src/widgets/async.dart

```diff
@@ -0,0 +1,256 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Widgets that handle interaction with asynchronous computations.
+
+import 'package:flutter/widgets.dart';
+import 'dart:async' show Future, Stream, StreamSubscription;
+
+/// Snapshot of an asynchronous interaction.
+class AsyncSnapshot<T> {
+  /// Current state of connection to the asynchronous computation.
+  final ConnectionState connectionState;
+
+  /// Latest data received. Is null, if [error] is not.
+  final T data;
+
+  /// Latest error object received. Is null, if [data] is not.
+  final dynamic error;
+
+  AsyncSnapshot(this.connectionState, this.data, this.error);
+
+  bool get hasError => error != null;
+  bool get hasData => data != null;
+
+  @override
+  String toString() => 'AsyncSnapshot($connectionState, $data, $error)';
+
+  @override
+  bool operator ==(dynamic o) =>
```

Comment:
nit: please use the same style for operator == as other existing operator ==s in the code base.